src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1050,23 +1169,6 @@ public bool IsTopMostNode
             }
         }
 
-        internal void ConnectInput(int inputData, int outputData, NodeModel node)
```

Comment:
so if we're talking about the same thing - *which as Greg said, I'm not sure the caches you're talking about are the same as the caches I am referring to in the pr I linked*.

To summarize the PR description: The traversal is slow during graph load, during freeze change, during watch3d connection, because every node is potentially performing a graph traversal to the root of the graph.

Graph load is even slower because as all properties are set on each node while it loads and connectors are added this fires off multiple graph traversals per node, per connector, per event...